src/hooks/useDatasetManager.js

```diff
@@ -68,8 +68,10 @@ export const useDatasetManager = () => {
   }
 
   const startProcessingDataset = async (id, downloadOptions) => {
-    // create a new token if none
-    const tokenId = token || (await createToken())
+    // validate the existing token or create a new token if none
+    const tokenId = !validateToken()
+      ? await resetToken()
+      : token || (await createToken())
```

Comment:
I think this would be easier to see what this evaluates to if we didn't combine a ternary operator and short-circuiting. Also, since we should change the order to handle "truthy" first.

Reply:
@davidsmejia, actually I had an error in this logic 😅 I've updated the ternary operator and is no longer uses a short-circuiting 👍